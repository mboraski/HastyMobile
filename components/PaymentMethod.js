// 3rd Party Libraries
import React from 'react';
import { StyleSheet, Text, TouchableOpacity, Image } from 'react-native';

// Relative Imports
import CardImage from '../components/CardImage';
import Color from '../constants/Color';
import Style from '../constants/Style';
import { emY } from '../utils/em';
import icon from '../assets/icons/keyboard-arrow-right.png';

export default function PaymentMethod({ type, text, style, ...props }) {
    return (
        <TouchableOpacity {...props} style={[styles.container, style]}>
            {type ? <CardImage type={type} style={styles.card} /> : null}
            <Text style={styles.text}>{text}</Text>
            <Image source={icon} style={styles.icon} />
        </TouchableOpacity>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        paddingHorizontal: 12,
        paddingVertical: 15,
        backgroundColor: Color.GREY_100,
        borderColor: Color.GREY_300,
        borderBottomWidth: StyleSheet.hairlineWidth,
    },
    card: {
        marginRight: 10
    },  
    text: {
        flex: 1,
        fontSize: emY(1),
    },
    icon: {
        
    }
});
