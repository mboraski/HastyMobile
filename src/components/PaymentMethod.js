// 3rd Party Libraries
import React from 'react';
import { StyleSheet, TouchableOpacity, Image } from 'react-native';

// Relative Imports
import Text from './Text';
import CardImage from '../components/CardImage';
import Color from '../constants/Color';
// import Style from '../constants/Style';
import { emY } from '../utils/em';
import icon from '../assets/icons/keyboard-arrow-right.png';

export default function PaymentMethod({ type, brand, last4, style, ...props }) {
    return (
        <TouchableOpacity {...props} style={[styles.container, style]}>
            {type ? (
                <CardImage type={type.toLowerCase()} style={styles.card} />
            ) : null}
            <Text style={styles.text}>{brand}</Text>
            <Text style={styles.cardNumbers}>**** {last4}</Text>
            <Image source={icon} />
        </TouchableOpacity>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        paddingHorizontal: 10,
        paddingVertical: 10,
        marginBottom: 10,
        backgroundColor: Color.GREY_100,
        borderColor: Color.GREY_300,
        borderTopWidth: StyleSheet.hairlineWidth,
        borderBottomWidth: StyleSheet.hairlineWidth
    },
    card: {
        marginRight: 10
    },
    cardTypeText: {
        flex: 1,
        fontSize: emY(1.1)
    },
    cardNumbers: {
        flex: 1,
        fontSize: emY(1.1),
        textAlign: 'right',
        marginRight: 15,
        color: Color.GREY_800
    }
});
