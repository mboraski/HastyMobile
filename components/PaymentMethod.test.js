import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import PaymentMethod from './PaymentMethod';

describe('PaymentMethod', () => {
    it('renders correctly', () => {
        const render = shallow(<PaymentMethod />);
        expect(render).toMatchSnapshot();
    });
});
