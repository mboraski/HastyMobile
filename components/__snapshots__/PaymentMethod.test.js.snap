// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethod renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    style={
        Array [
            Object {
              "backgroundColor": "#F4F4F4",
              "borderBottomWidth": 0.5,
              "borderColor": "#cccccc",
              "flexDirection": "row",
              "paddingHorizontal": 12,
              "paddingVertical": 15,
            },
            undefined,
          ]
    }
>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "flex": 1,
                "fontSize": 16,
              }
        }
    />
    <Image
        source={1}
        style={Object {}}
    />
</TouchableOpacity>,
  "nodes": Array [
    <TouchableOpacity
      activeOpacity={0.2}
      focusedOpacity={0.7}
      style={
            Array [
                  Object {
                    "backgroundColor": "#F4F4F4",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#cccccc",
                    "flexDirection": "row",
                    "paddingHorizontal": 12,
                    "paddingVertical": 15,
                  },
                  undefined,
                ]
      }
>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                  Object {
                        "flex": 1,
                        "fontSize": 16,
                      }
            }
      />
      <Image
            source={1}
            style={Object {}}
      />
</TouchableOpacity>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <PaymentMethod />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          style={
                    Array [
                              Object {
                                "backgroundColor": "#F4F4F4",
                                "borderBottomWidth": 0.5,
                                "borderColor": "#cccccc",
                                "flexDirection": "row",
                                "paddingHorizontal": 12,
                                "paddingVertical": 15,
                              },
                              undefined,
                            ]
          }
>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "flex": 1,
                                        "fontSize": 16,
                                      }
                    }
          />
          <Image
                    source={1}
                    style={Object {}}
          />
</TouchableOpacity>,
        "_debugID": 2,
        "_renderedOutput": <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          style={
                    Array [
                              Object {
                                "backgroundColor": "#F4F4F4",
                                "borderBottomWidth": 0.5,
                                "borderColor": "#cccccc",
                                "flexDirection": "row",
                                "paddingHorizontal": 12,
                                "paddingVertical": 15,
                              },
                              undefined,
                            ]
          }
>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "flex": 1,
                                        "fontSize": 16,
                                      }
                    }
          />
          <Image
                    source={1}
                    style={Object {}}
          />
</TouchableOpacity>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PaymentMethod />,
}
`;
